test/gtest/common/mem_buffer.cc

```diff
@@ -202,41 +202,63 @@ void mem_buffer::pattern_fill(void *buffer, size_t length, uint64_t seed)
     memcpy(ptr, &seed, end - (char*)ptr);
 }
 
-void mem_buffer::pattern_check(const void *buffer, size_t length, uint64_t seed)
+void mem_buffer::pattern_check(uint64_t expected, uint64_t actual,
+                               size_t length, size_t offset, const void *buffer,
+                               const void *orig_ptr)
 {
-    const char* end = (const char*)buffer + length;
+    const uint64_t mask = UCS_MASK_SAFE(length * 8 * sizeof(char));
```

Comment:
IIUC it should be like this?
```suggestion
    const uint64_t mask = UCS_MASK_SAFE(length * sizeof(expected));
```

Reply:
any specific reason why it's  `8 * sizeof(char)` but not `sizeof(expected)`?

Reply:
yes: it's we need to convert length (in bytes) to mask (in bits), so it's * 8. 
event if expected was "uint32_t" we'd still need to use * 8